tests/integration/endpoint_test.go

```diff
@@ -38,15 +39,42 @@ func endpointTests(t *testing.T, when spec.G, it spec.S) {
 
 			assert.Equal(t, "Hello World", service.GetEndpointResponse())
 			assert.Equal(t, "Hello World v3", stagedService.GetEndpointResponse())
-		})
-		it("should have the service app endpoint properly created", func() {
 
-			// Check the hostnames returned by Rio and Kubectl are equal
 			assert.Equal(t,
 				testutil.GetHostname(service.GetKubeAppEndpointURL()),
 				testutil.GetHostname(service.GetAppEndpointURL()),
 			)
+			assert.Equal(t,
+				testutil.GetHostname(service.GetAppEndpointURL()),
+				testutil.GetHostname(stagedService.GetAppEndpointURL()),
+			)
 			assert.Equal(t, "Hello World", service.GetAppEndpointResponse())
 		})
 	}, spec.Parallel())
+
+	when("a staged service is promoted", func() {
+		it.Before(func() {
+			service.Create(t, "ibuildthecloud/demo:v1", "-p", "80/http")
+			stagedService = service.Stage("ibuildthecloud/demo:v3", "v3")
+			stagedService.Promote("--rollout=false")
+		})
+		it.After(func() {
+			service.Remove()
+			stagedService.Remove()
+		})
+
+		it("should retain all revision endpoints with an app endpoint pointing to the new revision", func() {
+			assert.Equal(t, "Hello World", service.GetEndpointResponse())
+			assert.Equal(t, "Hello World v3", stagedService.GetEndpointResponse())
+			assert.Equal(t, 0, service.GetKubeCurrentWeight())
+			assert.Equal(t, 100, stagedService.GetKubeCurrentWeight())
+			assert.Equal(t, "Hello World v3", service.GetAppEndpointResponse())
+		})
+		it("should allow rolling back to the previous revision", func() {
+			service.Promote("--rollout-increment", "25", "--rollout-interval", "2")
```

Comment:
Doesn't this increment/interval screw up the test ? How does it get to 100 immediately.

Reply:
The promote function waits for weight to be 100%. I'm glad you brought this up. I think maybe for integration tests it makes sense to not have a rollout, and we can (and do) validate rollout in other tests. I'll change this one.